### PR TITLE
Pockets

### PR DESCRIPTION
**Pull request by fadedDexofan**
:cl: Pockets
add: Added pockets system.
wip: Pockets in suits.
/:cl:
To add a pocket of a suit, you need to change the path. For example:
Old path: /obj/item/clothing/suit/det_suit
New path: /obj/item/clothing/suit/storage/det_suit

### CODE REVIEW

**Comment by optimumtact** [comment]
Don't we already have pockets?

**Comment by dexofan23** [comment]
No pockets in clothes we have

**Comment by WJohn** [comment]
I think he means suit storage, which various clothes have, such as labcoats, armor vests, rig suits and so on.

**Comment by dexofan23** [comment]
547d7d7191c4412bbf47b3ccfa093540

File: code/game/objects/items/weapons/storage/internal.dm [hunk 1]
```diff
@@ -1,42 +1,80 @@
...
+	verbs -= /obj/item/verb/verb_pickup
 	..()
-	if(loc) name = loc.name
 
-/obj/item/weapon/storage/internal/pocket/big
-	max_w_class = 3
-	max_combined_w_class = 6
+/obj/item/weapon/storage/internal/Destroy()
+	master_item = null
+	..()
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/storage/internal.dm]: return ..()

File: code/modules/clothing/suits/storage.dm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	var/obj/item/weapon/storage/internal/pockets/pockets
+
+/obj/item/clothing/suit/storage/New()
+	..()
+	pockets = new/obj/item/weapon/storage/internal/pockets(src, slots = 2, slot_size = 2) //two slots, fit only pocket sized items
+
+/obj/item/clothing/suit/storage/Destroy()
+	qdel(pockets)
+	pockets = null
+	..()
```
Comment (ChangelingRain) [inline code/modules/clothing/suits/storage.dm]: return ..()

**Comment by anconfuzedrock** [comment]
fuck yea this was the raddest shit when I tried polaris and my military jacket had 2 fucking pockets I mean holy fuck it was awesome

**Comment by dexofan23** [comment]
Pockets size is customizable

**Comment by Zilenan91** [comment]
Heck yeah :+1:

**Comment by kevinz000** [comment]
breddy gud

**Comment by Armhulen** [comment]
Me like.

**Comment by WJohn** [comment]
okay but then why do we still have a suit storage slot??

**Comment by Core0verload** [comment]
Basically pockets from Animus, but with more snowflake types. Don't you mind me opening my own PR?

**Comment by coiax** [comment]
The mob strip code is gonna need more work to support these pockets.

**Comment by dexofan23** [comment]
@Core0verload, pockets from Baystation12. It's adapted port.

**Comment by FlavoredCactus** [comment]
Would work better if you put storage support in all suits editable with a variable. say... "pockets = 2"
Having to require people change paths will often not end well especially if said items are on the map. map edits would be frequent and given that we have like 30 maps including ruins, shuttles, and templates this would be a mess to maintain.

**Comment by coiax** [comment]
we've got automatic checking on all maps in our repo, so someone can't break the path of an object without it being noticed

**Comment by FlavoredCactus** [comment]
It is still more efficient especially when you have subtypes of say.. armor that need pockets and instead of copy pasting armor's stats it would just be a child of it while still having pockets.

**Comment by dexofan23** [comment]
I try to implement it as a suits property

**Comment by dexofan23** [comment]
Updated

**Comment by optimumtact** [comment]
Hi, I apologise for the delay in getting back to this.

Thanks for your contribution, but #19237 seems to me to be a cleaner implementation that can be applied to more items, even if I do disagree with the way the PR creator went about creating it.

**Comment by Chocobro** [comment]
>11:03:24 PM <oranges>  I feel bad for the creator of #19205

**Pull request closed without merging**